src/main/scala/com/timeout/KinesisSource.scala

```diff
@@ -1,41 +1,84 @@
 package com.timeout
 
 import java.nio.ByteBuffer
-import java.time.ZonedDateTime
+import java.time.{Clock, ZonedDateTime}
 import java.util.Date
 import java.util.concurrent.{Future => JavaFuture}
 
+import akka.NotUsed
 import akka.stream.scaladsl.Source
 import akka.stream.stage.{GraphStage, OutHandler, StageLogging, TimerGraphStageLogic}
 import akka.stream.{Attributes, Outlet, SourceShape}
 import com.amazonaws.regions.{Region, Regions}
 import com.amazonaws.services.kinesis.AmazonKinesisAsyncClient
-import com.amazonaws.services.kinesis.model.{GetRecordsRequest, GetRecordsResult, GetShardIteratorRequest}
+import com.amazonaws.services.kinesis.model._
 
 import scala.collection.JavaConverters._
 import scala.collection.mutable
-import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
 import scala.util.Try
 
 object KinesisSource {
 
-  def apply(stream: String, since: ZonedDateTime)(implicit ec: ExecutionContext) =
+  private [timeout] val region: Region =
+    Option(Regions.getCurrentRegion)
+      .getOrElse(Region.getRegion(Regions.EU_WEST_1))
+
+  private[timeout] lazy val kinesis: AmazonKinesisAsyncClient =
+    new AmazonKinesisAsyncClient().withRegion(region)
```

Comment:
According to the doc the AmazonKinesisAsyncClientBuilder is preferred over using constructors